scipy/integrate/_quadrature.py

```diff
@@ -360,33 +361,29 @@ def _basic_simpson(y, start, stop, x, dx, axis):
     slice2 = tupleset(slice_all, axis, slice(start+2, stop+2, step))
 
     if x is None:  # Even-spaced Simpson's rule.
-        result = np.sum(y[slice0] + 4.0*y[slice1] + y[slice2], axis=axis)
-        result *= dx / 3.0
+        result = dx / 3.0 * xp.sum(y[slice0] + 4.0*y[slice1] + y[slice2], axis=axis)
     else:
         # Account for possibly different spacings.
         #    Simpson's rule changes a bit.
-        h = np.diff(x, axis=axis)
+        h = xp.diff(x, axis=axis)
         sl0 = tupleset(slice_all, axis, slice(start, stop, step))
         sl1 = tupleset(slice_all, axis, slice(start+1, stop+1, step))
-        h0 = h[sl0].astype(float, copy=False)
-        h1 = h[sl1].astype(float, copy=False)
+        h0 = h[sl0] if xp_size(h) else h[...]
+        h1 = h[sl1] if xp_size(h) else h[...]
```

Comment:
When `h` is size-zero, `array-api-strict` considers the slice invalid. There is nothing to return but the original, size-zero array.
```suggestion
        h0 = h[sl0] if xp_size(h) else h
        h1 = h[sl1] if xp_size(h) else h
```